### ISSUE TITLE
`reflection.class_name` is sometimes scoped under some namespace and causes constant resolution errors

### ISSUE THREAD

**Issue by manhhung741** [chanzuckerberg/sorbet-rails#15]
We should use `"::#{reflection.klass.name}"` instead of `reflection.class_name` to avoid the problem

https://github.com/chanzuckerberg/sorbet-rails/blob/6fcf542e663ebe015c50dc67f8e4338da020d620/lib/sorbet_rails/models_rbi_formatter.rb#L109

**Comment by XrXr** [comment]
I think I exaggerated this problem in my original message. I must've ran into the error after making modifications. Sorry! The problem is real, but it does not lead to constant resolution errors.

I made an example app to demonstrate the problem. It's nothing fancy, basically just `rails new` and `rails generate model blog/component`

There are two models in the app, Article and Comment, both of which are under the `Blog` namespace. In this case, `reflection.class_name` returns `"Article"` for the `belongs_to :article` association on `Blog::Comment`. sorbet-rails sees and generates T.untyped for the association getter setter, when it could've generated `::Blog::Article` if it used `"::#{reflection.klass.name}"`

https://github.com/XrXr/sorbet-rails-bug-repro/commit/207d3d83a4ab0c65ca2f3656eb268f34cb6ceeea
